File: src/main/java/org/app/infrastructure/endpoints/dto/GetFriendListDTO.java
package org.app.infrastructure.endpoints.dto;

import org.app.domain.events.FriendEvent;

import java.util.List;

public class GetFriendListDTO {
    public List<GetUserItemDTO> friends;
    public int page;
    public int limit;
    public int totalPage;

    public GetFriendListDTO(List<GetUserItemDTO> friends, int page, int limit, int totalPage) {
        this.friends = friends;
        this.page = page;
        this.limit = limit;
        this.totalPage = totalPage;
    }

    public GetFriendListDTO(FriendEvent event) {
        this.friends = event.getFriends().stream().map(GetUserItemDTO::new).toList();
        this.page = event.getPage();
        this.limit = event.getLimit();
        this.totalPage = event.getTotalPage();
    }

    public static GetFriendListDTO from(FriendEvent event){
        return new GetFriendListDTO(event);
    }
}

File: go/app/main/infrastructure/endpoints/dto/get_friend_list_dto.go
package dto

import (
    "mutualAttentionSystem/app/main/domain/events"
    "mutualAttentionSystem/app/main/infrastructure/utils"
)

type GetFriendListDTO struct {
    Friends   []*GetUserItemDTO
    Page      int
    Limit     int
    TotalPage int
}

func ToGetFriendListDTO(event *events.GetUserEvent, page int, limit int) (*GetFriendListDTO, error) {
    pageable, err := utils.NewPableable(page, limit, len(event.User.GetFollowings()))
    if err != nil {
        return nil, err
    }
    getFriendListDTO := GetFriendListDTO{make([]*GetUserItemDTO, 0), page, limit, pageable.TotalPage}
    for _, user := range event.User.GetFriends() {
        getFriendListDTO.Friends = append(getFriendListDTO.Friends, ToGetUserItem(user))
    }
    return &getFriendListDTO, nil
}
